web_application/Backend/FlaskApp/app.py

```diff
@@ -532,10 +591,127 @@ def reject_invitation():
             {'_id': invitation['_id']},
             {'$set': {'status': 'rejected'}}
         )
-        return jsonify({'message': 'Invitation status updated successfully to rejected'}), 200
+        collaboration_uuid = invitation.get('collaboration_uuid')
+        receiver_id = invitation.get('receiver_id')
+
+        if collaboration_uuid and receiver_id:
+            db.collaborations.update_one(
+                {'uuid': collaboration_uuid, 'invited_users.user_id': receiver_id},
+                {'$set': {'invited_users.$.status': 'rejected'}}
+            )
+
+        return jsonify({'message': 'Invitation and collaboration status updated successfully to withdrawn'}), 200
     else:
         return jsonify({'message': 'No matching invitation found'}), 404
 
+
+@app.route('/api/start_collaboration', methods=['POST'])
+def start_collaboration():
+    try:
+        current_user, error_response = get_current_user()
+        if error_response:
+            return error_response
+        creator_id = current_user.id
+
+        data = request.get_json()
+        collab_name = data.get('collabName')
+        experiments = data.get('experiments', [])
+        raw_data = data.get('rawData')
+        meta_data = data.get('metaData')
+        invited_users = data.get('invitedUsers', [])
+
+        if not collab_name:
+            logging.error("Collaboration name missing in the request")
+            return jsonify({"error": "Collaboration name is required"}), 400
+
+        collaboration = {
+            'uuid': str(uuid.uuid4()),
+            'name': collab_name,
+            'experiments': experiments,
+            'raw_data': raw_data,
```

Comment:
Should we put data and metadata inside 'invited_users' and make it per 'user'(collaborator)?